src/main/app/claims/converters/claimantResponseConverter.ts

```diff
@@ -37,27 +39,53 @@ export class ClaimantResponseConverter {
     if (draftClaimantResponse.formaliseRepaymentPlan) {
       respAcceptance.formaliseOption = this.getFormaliseOption(draftClaimantResponse.formaliseRepaymentPlan)
     }
-    const courtDetermination: CourtDetermination = draftClaimantResponse.courtDetermination
+    const courtDetermination: DomainCourtDetermination = this.getCourtDetermination(draftClaimantResponse.courtDetermination)
     if (courtDetermination) {
-      respAcceptance.courtDetermination = {
-        courtDecision: courtDetermination.courtDecision,
-        courtPaymentIntention: courtDetermination.courtPaymentIntention,
-        rejectionReason: courtDetermination.rejectionReason ?
-          courtDetermination.rejectionReason.text : undefined,
-        disposableIncome: courtDetermination.disposableIncome ?
-          courtDetermination.disposableIncome : 0,
-        decisionType: courtDetermination.decisionType
-      }
+      respAcceptance.courtDetermination = courtDetermination
     }
-    const claimantPaymentIntention: ModelPaymentIntention = draftClaimantResponse.alternatePaymentMethod
+    const claimantPaymentIntention: DomainPaymentIntention = this.getClaimantPaymentIntention(draftClaimantResponse)
     if (claimantPaymentIntention) {
-      respAcceptance.claimantPaymentIntention = claimantPaymentIntention.toDomainInstance()
+      respAcceptance.claimantPaymentIntention = claimantPaymentIntention
+    }
+    return respAcceptance
+  }
+
+  private static getCourtDetermination (courtDetermination: CourtDetermination): DomainCourtDetermination {
+    const decisionType: DecisionType = courtDetermination.decisionType
+    if (decisionType === DecisionType.COURT && !courtDetermination.courtPaymentIntention) {
+      throw new Error('court offered payment intention not found where decision type is COURT')
+    }
+    if (decisionType === DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT) {
+      return undefined
```

Comment:
no CourtDetermination when DecisionType.CLAIMANT_IN_FAVOUR_OF_DEFENDANT? please double check this...

Reply:
done. removed this check.